snuba/datasets/configuration/generic_metrics/migrations/0001_sets_aggregate_table.yaml

```diff
@@ -0,0 +1,130 @@
+version: v1
+kind: migration
+name: 0001_sets_aggregate_table
+
+forward_local:
+  - create_table
+    storage_set: generic_metrics_sets
+    table_name: generic_metric_sets_local
+
+    engine:
+      type: AggregatingMergeTree
+      order_by: "(org_id, project_id, metric_id, granularity, timestamp, tags.key, tags.indexed_value, tags.raw_value, retention_days, use_case_id)"
+      primary_key: "(org_id, project_id, metric_id, granularity, timestamp)"
+      partition_by: "(retention_days, toMonday(timestamp))"
+      settings:
+        index_granularity: 2048
+      ttl: "timestamp + toIntervalDay(retention_days)"
+
+    columns:
+      [
+        { name: org_id, type: UInt, args: { size: 64 } },
+        { name: project_id, type: UInt, args: { size: 64 } },
+        { name: metric_id, type: UInt, args: { size: 64 } },
+        { name: granularity, type: UInt, args: { size: 8 } },
+        { name: timestamp, type: DateTime, modifiers: { codecs: [DoubleDelta] } },
+        { name: retention_days, type: UInt, args: { size: 16 } },
+        {
+          name: tags,
+          type: Nested,
+          args:
+            {
+              subcolumns:
+                [
+                  { name: key, type: UInt, args: { size: 64 } },
+                  { name: indexed_value, type: UInt, args: { size: 64 } },
+                  { name: raw_value, type: String },
+                ],
+            },
+        },
+        {
+          name: value,
+          type: AggregateFunction,
+          args: { func: "uniqCombined64", arg_types: [{ type: UInt, arg: 64 }] },
+        },
+        { name: use_case_id, type: String, modifiers: { low_cardinality: true } },
+      ]
+
+  - add_column
+    storage_set: generic_metrics_sets
+    table_name: generic_metric_sets_local
+    column:
+      name: _indexed_tags_hash
+      type: Array
+      args: { type: UInt, arg: 64 }
+      modifiers: { materialized: "arrayMap((k, v) -> cityHash64(concat(toString(k), '=', toString(v))), tags.key, tags.indexed_value)" }
+    after: null
+
+  - add_column
+    storage_set: generic_metrics_sets
+    table_name: generic_metric_sets_local
+    column:
+      name: _raw_tags_hash
+      type: Array
+      args: { type: UInt, arg: 64 }
+      modifiers: { materialized: "arrayMap((k, v) -> cityHash64(concat(toString(k), '=', v)), tags.key, tags.raw_value)" }
+    after: null
```

Comment:
@onewland, fyi, these columns should be created on the distributed table as well. In fact they are missing in production.